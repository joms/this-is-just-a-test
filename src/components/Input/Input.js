import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

const Input = ({ label, placeholder, onChange, value, type, error, hint, ...rest }) => (
    <>
        <label>{label}</label>
        <input
            {...rest}
            onChange={onChange}
            type={type}
            placeholder={placeholder}
            value={value}
            className={cx({ error })}
        />
        {error && <span className="error">{error}</span>}
        {hint && <span className="hint">{hint}</span>}
    </>
);

Input.defaultProps = {
    placeholder: '',
    onChange: () => null,

    value: '',
    error: '',
    hint: '',
};

Input.propTypes = {
    label: PropTypes.string.isRequired,
    placeholder: PropTypes.string,
    onChange: PropTypes.func,
    value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    error: PropTypes.string,
    hint: PropTypes.string,
};

export default Input;
