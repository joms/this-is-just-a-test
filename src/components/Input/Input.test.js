import React from 'react';
import { shallow } from 'enzyme';
import Input from '.';

describe('<Input>', () => {
    const onChange = jest.fn();
    const wrapper = shallow(
        <Input label="Hello, input" onChange={onChange} hint="this is a hint" error="This field is required" />
    );

    it('matches snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('handles input change', () => {
        wrapper.find('input').simulate('change', 'Hi thar!');
        expect(onChange).toHaveBeenCalledWith('Hi thar!');
    });

    it('renders hint text', () => expect(wrapper.text().includes('this is a hint')).toBeTruthy());

    it('renders error text', () => expect(wrapper.text().includes('This field is required')).toBeTruthy());
});
