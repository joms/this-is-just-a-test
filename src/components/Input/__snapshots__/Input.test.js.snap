// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input> matches snapshot 1`] = `
<Fragment>
  <label>
    Hello, input
  </label>
  <input
    className="error"
    onChange={[MockFunction]}
    placeholder=""
    value=""
  />
  <span
    className="error"
  >
    This field is required
  </span>
  <span
    className="hint"
  >
    this is a hint
  </span>
</Fragment>
`;
